### ISSUE TITLE
Making static .zsh file doesn't work on shared directories

### ISSUE THREAD

**Issue by dberlin** [mattmc3/antidote#20]
I share my zsh config among a bunch of servers through a share.
This works fine with antibody, the bundles are cloned into a local directory, etc.

With antidote, it creates a zsh_plugins.zsh file in the same dir as zsh_plugins.txt.  Since this is a shared directory (in my case) , it gets overwritten by each machine that runs a shell :)
It would be nice to put this file in a local cache directory (or at least give the option to do so) rather than create it in the same dir as zsh_plugins.txt. That would make antidote work out of the box in my case, as antibody does.

**Comment by mattmc3** [comment]
Can you post the code you're using? This should work as-is, but maybe I'm not understanding your issue.

Antidote was designed for feature parity with antibody, so if you use it the same way you used antibody, it should all work the same way. Antidote also includes some new commands antibody didn't have like the `load` command, which already supports an argument for the plugins file:

```zsh
antidote load ~/.zsh.server1/server1_plugins.txt
```

You can also call `antidote bundle` the same way that `antibody` worked and export your static file wherever you want it:

```zsh
antidote bundle < ~/.zsh_plugins.txt > ~/.zsh_plugins.sh
```

**Comment by dberlin** [comment]
Antibody code was the following:
```
source <(antibody init)
antibody bundle <~/.zsh/antibody_plugins.txt

```
Imagine the following:

~/.zsh is a shared directory on an NFS server.  All machines mount it. (it's really dropbox but it doesn't matter)
WIth antibody, the above will create no files in ~/.zsh. Zero.

With antidote, used like this:
```
# clone antidote if necessary
#[[ -e ~/.antidote ]] || git clone https://github.com/mattmc3/antidote.git ~/.antidot

# source antidote
# . ~/.antidote/antidote.zsh
#
# generate and source plugins from ~/.zsh_plugins.txt
#antidote load ~/.zsh/antibody_plugins.txt

```

This will create a ~/.zsh/antibody_plugins.zsh
It will then try to load this file next time.
This file is in the shared directory, and everything that has that dir mounted will try to use it.
However, that file has paths specific to whatever machine generated it.
Which won't work.

With antibody, because it does not create a .zsh file in the shared directory, i don't have to do anything.

With antibody, it looks like i have to manually manage a local directory and cache with bundle/load?

**Comment by mattmc3** [comment]
Looks like you were using the dynamic config option with Antibody. Antidote supports that method too.

So, if you do this, you won't get that static file:

```zsh
source <(antidote init)
antidote bundle <~/.zsh/antidote_plugins.txt
```

Will that work for your use case? Since it doesn't create the static file and just evals the generated Zsh code, you don't get the highest performance config that way, but you didn't with antibody either and that may be fine for you. Let me know if that doesn't work and we can dig in further.

**Comment by dberlin** [comment]
Yes, you are totally right, i wasn't thinking.  Doing it this way works. It loads in an immeasurably small amount of time, so i'm not too worried.  

I assume to get the fastest I can use something like
```
# clone antidote if necessary
[[ -e ~/.antidote ]] || git clone https://github.com/mattmc3/antidote.git ~/
.antidote
source ~/.antidote/antidote.zsh

LOCAL_CACHE_BASE=${XDG_CACHE_HOME:-$HOME/.cache}/antidote-$HOST
[[ -e ${LOCAL_CACHE_BASE} ]] ||  mkdir -p ${LOCAL_CACHE_BASE}

[[ ~/.zsh/antibody_plugins.txt -nt ${LOCAL_CACHE_BASE}/antibody_plugins.zsh
]] || antidote bundle <~/.zsh/antibody_plugins.txt >${LOCAL_CACHE_BASE}/anti
body_plugins.zsh
source ${LOCAL_CACHE_BASE}/antibody_plugins.zsh
```
and it will work.

It looks like load  does something like this, but it doesn't look like i can use it directly because you can't redirect where it places the staticfile

**Comment by mattmc3** [comment]
You are correct. I'm going to leave this one open because there may be an opportunity here for a couple enhancements:

* Update the docs to reflect the dynamic load method exists, similar to antibody's
* Update the `antidote load` method to take an output file as a second parameter

**Comment by dberlin** [comment]
Sounds good. Thanks for your  help, it is really appreciated.  If you change the load method and want someone to try it out, happy to volunteer!

**Comment by mattmc3** [comment]
The second param now works and docs updated: https://getantidote.github.io/migrating-from-antibody
